src/Orleans.TestingHost/TestCluster.cs

```diff
@@ -138,6 +139,40 @@ public void Deploy()
                 string startMsg = "----------------------------- STARTING NEW UNIT TEST SILO HOST: " + GetType().FullName + " -------------------------------------";
                 WriteLog(startMsg);
                 await InitializeAsync();
+
+                // Poll each silo to check that it knows the expected number of active silos.
+                // If any silo does not have the expected number of active silos in its cluster membership oracle, try again.
+                // If the cluster membership has not stabilized after a certain period of time, give up and continue anyway.
+                var totalWait = Stopwatch.StartNew();
```

Comment:
can you capsulate this section of code into a function `WaitForClusterStabilizing` or `MakeSureAllSiloJoinedCluster` method. Just to make it cleaner and more readable